scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -136,6 +144,53 @@ def generate_stat_var_map(spec_dict, column_list, delimiter='!!'):
     return col_map_obj._generate_stat_vars_from_spec()
 
 
+def generate_mcf_from_column_map(column_map: dict,
+                                 output_path_dir: str = './') -> str:
+    """Function that generates and stores mcf file from given column map
+    Args:
+        column_map: dict containing yearwise mapping of column name to it's statvar
+        output_path_dir: path at which the output file is to be stored
+    Returns:
+        str containing the statvar definitions, this is the string written to output file (statvars_all.mcf)
+    """
+    if not os.path.exists(output_path_dir):
+        os.makedirs(output_path_dir, exist_ok=True)
+
+    mcf_dict = {}
+
+    for year in column_map:
+        for column_name in column_map[year]:
+            dcid = column_map[year][column_name]['Node']
+            if dcid not in mcf_dict:
+                ## key --> node dcid
+                mcf_dict[dcid] = {}
+                ## add pvs to dict
+                for key, value in column_map[year][column_name].items():
+                    mcf_dict[dcid][key] = value
+
+    ## create final output string to be written to mcf file
+    final_mcf = ""
+
+    for dcid, stat_var in mcf_dict.items():
+        col_stat_var = []
+
+        dcid = 'Node: ' + dcid
+        col_stat_var.append(dcid)
+
+        for p, v in stat_var.items():
+            if p != 'unit' and p != 'scalingFactor' and p != 'Node':
+                col_stat_var.append(f"{p}: {v}")
+
+        final_mcf = final_mcf + '\n'.join(col_stat_var) + "\n\n"
+
+    print('Writing mcf output to', output_path_dir)
+    f = open(os.path.join(output_path_dir, 'statvars_all.mcf'), 'w')
+    f.write(final_mcf)
+    f.close()
+
+    return final_mcf
+
```

Comment:
Having the same function repeated in two places, is a bit difficult to maintain, perhaps, we can move this function to a common module?

Or better is to move this function to `process.py` since the column map is returned as a dictionary to process.py from `common/generate_col_map.py` module